main.go

```diff
@@ -115,21 +115,19 @@ func handleSigterm(stopChan chan struct{}) {
 	close(stopChan)
 }
 
-func newClient() (*kubernetes.Clientset, error) {
+func newClient(cfg *config.Config) (*kubernetes.Clientset, error) {
 	var (
 		config *rest.Config
 		err    error
 	)
 
-	kubeconfig := clientcmd.RecommendedHomeFile
-
-	// if inCluster {
-	// config, err = rest.InClusterConfig()
-	// log.Debug("Using in-cluster config.")
-	// } else {
-	config, err = clientcmd.BuildConfigFromFlags("", kubeconfig)
-	log.Debugf("Using current context from kubeconfig at %s.", kubeconfig)
-	// }
+	if cfg.InCluster {
+		config, err = rest.InClusterConfig()
+		log.Debug("Using in-cluster config.")
+	} else {
+		config, err = clientcmd.BuildConfigFromFlags("", cfg.KubeConfig)
```

Comment:
IMO we can have `--in-cluster` flag defaults to `true`.  We also have --kubeconfig flag with default value of "". Then in configuration file we would check if `--in-cluster` is set to `false`, then if cfg.KubeConfig is not set, then set it to `clientcmd.RecommendedHomeFIle`. But this is just nitpicking, I don't mind having the current implementation :)